app/controllers/api/v2/preupgrade_reports_controller.rb

```diff
@@ -3,17 +3,34 @@
 module Api
   module V2
     class PreupgradeReportsController < ::Api::V2::BaseController
-      before_action :find_resource, only: %i[show]
+      include ApiAuthorizer
 
       api :GET, '/preupgrade_reports/', N_('List Preupgrade reports')
       param_group :search_and_pagination, ::Api::V2::BaseController
       def index
-        @preupgrade_reports = resource_scope_for_index
+        @preupgrade_reports = PreupgradeReport.search_for(*search_options).paginate(paginate_options)
       end
 
       api :GET, '/preupgrade_reports/:id', N_('Show Preupgrade report')
       param :id, :identifier, required: true
-      def show; end
+      def show
+        @preupgrade_report = PreupgradeReport.find(params[:id])
+      end
+
+      api :GET, '/job_invocations/:id/preupgrade_reports', N_('List Preupgrade reports for Job invocation')
+      param :id, :identifier, required: true
+      def job_invocation
+        @preupgrade_reports = PreupgradeReport.where(job_invocation_id: params[:id])
```

Comment:
A job is run against hosts `A` and `B`. As a user with `view_job_invocations` and `view_hosts` scoped to host `A` I make a request to this endpoint? Will I be able to see preupgrade reports for both hosts?

Reply:
This should solve the issue:
```
hosts = JobInvocation.find(params[:id]).targeting.hosts.authorized(:view_hosts, Host)
@preupgrade_reports = PreupgradeReport.where(job_invocation_id: params[:id], host_id: hosts.ids)
```

Question is do we want to do this check also for `:show` and `:index` action?

Reply:
I would say yes. The users should be able to see only what they're allowed to see